### PR TITLE
add center / spacer / hstack / vstack in radix/themes/layout

### PR DESCRIPTION
**Pull request by Lendemor**
```python
import reflex.components.radix.themes as rdxt

rdxt.center(
    rdxt.hstack(
        rx.text("A"),
        rx.text("B"),
        rdxt.spacer(),
        rx.text("C"),
    )
)

```

### CODE REVIEW

File: reflex/components/core/layout/center.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+from reflex.components.component import Component
+from reflex.components.el.elements.typography import Div
+from reflex.style import Style
+
+
+class Center(Div):
+    """A center component."""
+
+    tag = "div"
```
Comment (masenf) [inline reflex/components/core/layout/center.py:11]: does this even need to be set if you inherit from `Div`?

File: reflex/components/core/layout/stack.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        Args:
+            *children: The children of the stack.
+            **props: The properties of the stack.
+
+        Returns:
+            The stack component.
+        """
+        style = props.setdefault("style", {})
+        align = props.pop("align", "center")
+        spacing = props.pop("spacing", "0.5rem")
```
Comment (masenf) [inline reflex/components/core/layout/stack.py:36]: where does the 0.5rem come from? just what "feels nice"?

more broadly, extending these from `Div` and not radix `Flex` means we dont get the common layout and common margin props. I can see the benefits of both ways, but as it is now, if you want consistency with radix themes components, then you probably end up reimplementing these yourself anyway, but based on radix `Flex` or `Box`

File: reflex/components/core/layout/stack.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    """A stack component."""
+
+    # The direction of the stack.
+    justify: Var[Literal["start", "center", "end"]]
+
+    # The alignment of the stack.
+    align: Var[Literal["start", "center", "end", "stretch"]]
+
+    # The spacing between each stack item.
+    spacing: Var[str]
```
Comment (masenf) [inline reflex/components/core/layout/stack.py:21]: a bit unexpected that these are given as props, considering they wouldn't render correctly if they were set after the fact (outside of create).

i don't think anything should be a prop unless it is expected to be rendered into the tag for the frontend code... there are some exceptions throughout the codebase, but in those cases, we explicitly strip out the "fake props" inside `_render`

File: reflex/components/core/layout/stack.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+
+class HStack(Stack):
+    """A horizontal stack component."""
+
+    def _apply_theme(self, theme: Component | None):
+        self.style.update({"flex_direction": "row"})
+
+
+# class Stack(rdxt.Flex):
```
Comment (masenf) [inline reflex/components/core/layout/stack.py:63]: 🤔 maybe _this_ stack should go under the radix themes hierarchy?

File: reflex/components/core/layout/stack.py (hunk 2 above)
Comment (Lendemor) [reply reflex/components/core/layout/stack.py:36]: same default as chakra for the "0.5rem"

File: reflex/components/core/layout/center.py (hunk 1 above)
Comment (Lendemor) [reply reflex/components/core/layout/center.py:12]: no i'll remove it.

File: reflex/components/core/layout/stack.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        """
+        style = props.setdefault("style", {})
+        style.update(
+            {
+                "display": "flex",
+                "alignItems": align,
+                "justifyContent": justify,
+                "gap": spacing,
+            }
+        )
```
Comment (masenf) [inline reflex/components/core/layout/stack.py:45]:
```suggestion
        props["style"] = {
                "display": "flex",
                "alignItems": align,
                "justifyContent": justify,
                "gap": spacing,
                **style,
        }
```

we should allow the user to override the styles right?

**Review by masenf — CHANGES_REQUESTED** [review]

**Review by picklelo — APPROVED** [review]

**Pull request merged**